src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -163,24 +165,39 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
-        public static unsafe double CopySign(double x, double y)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
-
-            long xbits = BitConverter.DoubleToInt64Bits(x);
-            long ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.IsSupported || AdvSimd.IsSupported)
+            {
+                var xvec = Vector128.CreateScalarUnsafe(x);
+                var yvec = Vector128.CreateScalarUnsafe(y);
 
-            // If the sign bits of x and y are not the same,
-            // flip the sign bit of x and return the new value;
-            // otherwise, just return x
+                var mask = Vector128.CreateScalarUnsafe(-0.0); // has sign bit set, everything else 0
 
-            if ((xbits ^ ybits) < 0)
+                return Vector128.ConditionalSelectBitwise(mask, yvec, xvec).ToScalar();
+            }
+            else
             {
-                return BitConverter.Int64BitsToDouble(xbits ^ long.MinValue);
+                return SoftwareFallback(x, y);
             }
 
-            return x;
+            static double SoftwareFallback(double x, double y)
+            {
+                const long signMask = 1L << 63;
+
+                // This method is required to work for all inputs,
+                // including NaN, so we operate on the raw bits.
+                long xbits = BitConverter.DoubleToInt64Bits(x);
+                long ybits = BitConverter.DoubleToInt64Bits(y);
+
```

Comment:
How does the perf of this implementation compare to the original one on x86?

Reply:
> Across the board, intrinsic is consistently fastest, and the branchless impl is fastest on all except Scenarios.Same for float, and Scenarios.Same and Scenarios.Alternating for double. However, it is <10us slower in these cases, and still >3x faster in Scenarios.Random, which is probably the most common scenario, so at the moment this PR uses the branchless impl for software fallback

Reply:
I think the question was explicitly with regards to 32-bit x86, rather than x64. The numbers will need to be collected independently.

Reply:
for random/unpredictable it is ~175us -> ~50us

Reply:
Could you share the before/after disasm for both x86 and x64?

Reply:
x64 - branched is original, branchless is new
```
CopySign_Branched(Double, Double)
    L0000: sub rsp, 0x18
    L0004: vzeroupper
    L0007: vmovsd [rsp+0x10], xmm0
    L000d: mov rax, [rsp+0x10]
    L0012: vmovsd [rsp+0x8], xmm1
    L0018: mov rdx, [rsp+0x8]
    L001d: xor rdx, rax
    L0020: test rdx, rdx
    L0023: jge L0040
    L0025: mov rdx, 0x8000000000000000
    L002f: xor rax, rdx
    L0032: mov [rsp], rax
    L0036: vmovsd xmm0, qword [rsp]
    L003b: add rsp, 0x18
    L003f: ret
    L0040: add rsp, 0x18
    L0044: ret


CopySign_Branchless(Double, Double)
    L0000: sub rsp, 0x18
    L0004: vzeroupper
    L0007: vmovsd [rsp+0x10], xmm0
    L000d: mov rax, [rsp+0x10]
    L0012: vmovsd [rsp+0x8], xmm1
    L0018: mov rdx, [rsp+0x8]
    L001d: mov rcx, 0x8000000000000000
    L0027: and rdx, rcx
    L002a: mov rcx, 0x7fffffffffffffff
    L0034: and rax, rcx
    L0037: or rax, rdx
    L003a: mov [rsp], rax
    L003e: vmovsd xmm0, qword [rsp]
    L0043: add rsp, 0x18
    L0047: ret


CopySign_Branched(Single, Single)
    L0000: sub rsp, 0x18
    L0004: vzeroupper
    L0007: vmovss [rsp+0x14], xmm0
    L000d: mov eax, [rsp+0x14]
    L0011: vmovss [rsp+0x10], xmm1
    L0017: mov edx, [rsp+0x10]
    L001b: xor edx, eax
    L001d: test edx, edx
    L001f: jge L0035
    L0021: xor eax, 0x80000000
    L0026: mov [rsp+0xc], eax
    L002a: vmovss xmm0, dword [rsp+0xc]
    L0030: add rsp, 0x18
    L0034: ret
    L0035: add rsp, 0x18
    L0039: ret

CopySign_Branchless(Single, Single)
    L0000: sub rsp, 0x18
    L0004: vzeroupper
    L0007: vmovss [rsp+0x14], xmm0
    L000d: mov eax, [rsp+0x14]
    L0011: vmovss [rsp+0x10], xmm1
    L0017: mov edx, [rsp+0x10]
    L001b: and edx, 0x80000000
    L0021: and eax, 0x7fffffff
    L0026: or eax, edx
    L0028: mov [rsp+0xc], eax
    L002c: vmovss xmm0, dword [rsp+0xc]
    L0032: add rsp, 0x18
    L0036: ret
```